api/commands/validate.ts

```diff
@@ -0,0 +1,50 @@
+import * as cronParser from "cron-parser";
+import { dataform } from "@dataform/protos";
+
+
+function validateEmail(email: string) {
+  const email_regex = /^(([^<>()\[\]\.,;:\s@\"]+(\.[^<>()\[\]\.,;:\s@\"]+)*)|(\".+\"))@(([^<>()[\]\.,;:\s@\"]+\.)+[^<>()[\]\.,;:\s@\"]{2,})$/i;
+  return email_regex.test(email);
+}
+
+export function validateSchedules(schedules: dataform.schedules.Schedules) {
+
+  if (schedules.defaultNotification && schedules.defaultNotification.emails) {
```

Comment:
pretty sure that `schedules.defaultNotification.emails` is *always* truthy, because it's a proto3 repeated field and thus is always at least an empty array.

Reply:
if I don't provide the defaultNotification data, it get undefined.

Reply:
I think there's some confusion here, sorry for that!

previously when I said to remove the proto3 default values, I didn't mean to remove setting a `defaultNotification` object - we'll presumably still need that, if it's a required field. I just meant that the fields *inside* that object don't need to be set explicitly if we're just going to set them to their default values, because that's a no-op.

note that if `defaultNotification` is indeed a required field, we should check that it is set in this validate function.

separately, at this line of code, if you only remove `schedules.defaultNotification.emails`, the code should function exactly the same. because of the way proto3 works, if `schedules.defaultNotification` evaluates to `true` (i.e. it is not `undefined`), `schedules.defaultNotification.emails` will *always* evaluate to `true`, because a proto3 repeated field is always at least an empty array.

Reply:
only checking the `defaultNotification` value now.